proto/wippersnapper/signal/v1/signal.proto

```diff
@@ -7,176 +7,86 @@ package wippersnapper.signal.v1;
-  oneof payload {
-    wippersnapper.ds18x20.v1.Ds18x20InitRequest req_ds18x20_init     = 1;
-    wippersnapper.ds18x20.v1.Ds18x20DeInitRequest req_ds18x20_deinit = 2;
-  }
-}
-
-/**
-* Ds18x20Response represents a Ds18x20 message from the device.
-*/
-message Ds18x20Response {
-  option (nanopb_msgopt).submsg_callback = true;
-  oneof payload {
-    wippersnapper.ds18x20.v1.Ds18x20InitResponse resp_ds18x20_init      = 1;
-    wippersnapper.ds18x20.v1.Ds18x20DeviceEvent  resp_ds18x20_event     = 2;
-  }
-}
 
-/**
-* I2CRequest represents the broker's request for a specific i2c command to a device.
-*/
-message I2CRequest {
+message BrokerToDevice {
   option (nanopb_msgopt).submsg_callback = true;
   oneof payload {
-    wippersnapper.i2c.v1.I2CBusInitRequest req_i2c_init                     = 1 [deprecated = true, (nanopb).type = FT_IGNORE];
-    wippersnapper.i2c.v1.I2CBusScanRequest req_i2c_scan                     = 2;
-    wippersnapper.i2c.v1.I2CBusSetFrequency req_i2c_set_freq                = 3;
-    wippersnapper.i2c.v1.I2CDeviceInitRequest req_i2c_device_init           = 4;
-    wippersnapper.i2c.v1.I2CDeviceDeinitRequest req_i2c_device_deinit       = 5;
-    wippersnapper.i2c.v1.I2CDeviceUpdateRequest req_i2c_device_update       = 6;
-    wippersnapper.i2c.v1.I2CDeviceInitRequests req_i2c_device_init_requests = 7;
+    // digitalio.proto
+    wippersnapper.digitalio.v1.DigitalIOAdd digitalio_add = 1;
+    wippersnapper.digitalio.v1.DigitalIOAdd digitalio_remove = 2;
+    wippersnapper.digitalio.v1.DigitalIOEvent digitalio_event = 3;
+    // analogio.proto
+    wippersnapper.analogio.v1.AnalogIOAdd analogio_add = 4;
+    wippersnapper.analogio.v1.AnalogIOAdd analogio_remove = 5;
+    // description.proto
+    wippersnapper.description.v1.CreateDescription create_description = 6;
+    // servo.proto
+    wippersnapper.servo.v1.ServoAdd servo_add = 7;
+    wippersnapper.servo.v1.ServoRemove servo_remove = 8;
+    wippersnapper.servo.v1.ServoWrite servo_write   = 9;
+    // pwm.proto
+    wippersnapper.pwm.v1.PWMAdd pwm_add                        = 10;
+    wippersnapper.pwm.v1.PWMRemove pwm_remove                        = 11;
+    wippersnapper.pwm.v1.PWMWriteDutyCycle pwm_write_duty            = 12;
+    wippersnapper.pwm.v1.PWMWriteDutyCycleMulti pwm_write_duty_multi = 13;
+    wippersnapper.pwm.v1.PWMWriteFrequency pwm_write_freq            = 14;
+    // pixels.proto
+    wippersnapper.pixels.v1.PixelsAdd pixels_add = 15;
+    wippersnapper.pixels.v1.PixelsRemove pixels_remove = 16;
+    wippersnapper.pixels.v1.PixelsWrite pixels_write   = 17;
+    // ds18x20.proto
+    wippersnapper.ds18x20.v1.Ds18x20Add ds18x20_add     = 18;
+    wippersnapper.ds18x20.v1.Ds18x20Remove ds18x20_remove = 19;
+    // uart.proto
+    wippersnapper.uart.v1.UARTAdd uart_add = 20;
+    wippersnapper.uart.v1.UARTRemove uart_remove = 21;
+    // i2c.proto
+    wippersnapper.i2c.v1.I2CScan i2c_scan = 22;
+    wippersnapper.i2c.v1.I2CInits i2c_add = 23;
```

Comment:
the scan is interesting, i think it's the only example of doing anything that isn't a component add/remove/event? I wonder if we'll have more of those kinds of messages over time, perhaps for battery, sleep, other functions down the road. 

`I2CInits`, maybe becomes `I2CInit` if we drop the repeated style as I suggested above in `i2c.proto`, and really should just be `I2CAdd`, yeah?

Reply:
> I2CInits, maybe becomes I2CInit if we drop the repeated style as I suggested above in i2c.proto, and really should just be I2CAdd, yeah?

Done